treasury-fees-monitor/src/utils.js

```diff
@@ -0,0 +1,70 @@
+const fs = require('fs');
+const csv = require('csv-parser');
+const BigNumber = require('bignumber.js');
+
+// helper function for importing a file located in the abi/ folder given its name
+function getAbi(abiName) {
+  // eslint-disable-next-line global-require,import/no-dynamic-require
+  const { abi } = require(`../abi/${abiName}`);
+  return abi;
+}
+
+function calculateStatistics(currMean, currVariance, currNumDataPoints, newValue) {
+  let newMean = 0;
+  let newStdDev = 0;
+  let newVariance = 0;
+  let newNumDataPoints = currNumDataPoints + 1;
+
+  if (currNumDataPoints === 0) {
+    newMean = newValue;
+    newNumDataPoints = 1;
+  } else {
+    newMean = (currMean * (currNumDataPoints / newNumDataPoints)) + (newValue / newNumDataPoints);
+    newVariance = (
+      (((currVariance * currNumDataPoints) + ((newValue - newMean) * (newValue - currMean)))
+        / newNumDataPoints)
+    );
+    newStdDev = Math.sqrt(newVariance);
+  }
+
+  return {
+    mean: newMean,
+    stdDev: newStdDev,
+    variance: newVariance,
+    numDataPoints: newNumDataPoints,
+  };
+}
+
+async function parseCsvAndCompute(csvFileName, tokenInfo, tokenPriceInfo) {
+  return new Promise((resolve, reject) => {
+    let mean = 0;
+    let stdDev = 0;
+    let variance = 0;
+    let numDataPoints = 0;
+
+    fs.createReadStream(`${__dirname}/${csvFileName}`)
+      .pipe(csv())
+      .on('data', (data) => {
+        // scale the premium for each row of data in the CSV file by the given asset's decimals
+        const denominator = (new BigNumber(10)).pow(tokenInfo[data.asset]);
+        const scaledPremium = parseFloat(
+          (new BigNumber(data.premium)).div(denominator),
+        );
+        const premiumInEth = scaledPremium * tokenPriceInfo[data.asset];
+
+        ({
+          mean, stdDev, variance, numDataPoints,
+        } = calculateStatistics(mean, variance, numDataPoints, premiumInEth));
+      })
+      .on('end', () => resolve({
+        mean, stdDev, variance, numDataPoints,
+      }))
+      .on('error', () => reject());
+  });
+}
```

Comment:
This BigNumber version is a _little_ bit slower, but it's the same order of magnitude as the parseFloat version, and the answer is slightly different:

The times belong are the returned values from `process.hrtime()`, so the format is:
seconds,nanoseconds

parseFloat version
```
numDataPoints: 2599
mean: 0.21372245107936824
variance: 30.398466619231904
stdDev: 5.5134804451663655
Total execution time: 0,63395292
```

BigNumber version
```
numDataPoints: 2599
mean: 0.21308075695371115401
variance: 30.20643640296592238208
stdDev: 5.49603824613383875805
Total execution time: 0,76689583
```

```suggestion
async function parseCsvAndCompute(csvFileName, tokenInfo, tokenPriceInfo) {
  return new Promise((resolve, reject) => {
    let denominator;
    let tokenPrice;
    const denominators = {};
    const tokenPrices = {};
    const premiumsInEth = [];

    fs.createReadStream(`${__dirname}/${csvFileName}`)
      .pipe(csv())
      .on('data', (data) => {
        // scale the premium for each row of data in the CSV file by the given asset's decimals
        denominator = denominators[data.asset];
        if (denominator === undefined) {
          denominator = new BigNumber(10).pow(tokenInfo[data.asset]);
          denominators[data.asset] = denominator;
        }

        tokenPrice = tokenPrices[data.asset];
        if (tokenPrice === undefined) {
          tokenPrice = new BigNumber(tokenPriceInfo[data.asset]);
          tokenPrices[data.asset] = tokenPrice;
        }

        const scaledPremium = new BigNumber(data.premium).div(denominator);
        const premiumInEth = scaledPremium.times(tokenPrice);
        premiumsInEth.push(premiumInEth);
      })
      .on('end', () => {
        const numDataPoints = premiumsInEth.length;
        const mean = BigNumber.sum(...premiumsInEth).div(numDataPoints);
        const squaredDifferences = premiumsInEth.map((sample) => sample.minus(mean).pow(2));
        const variance = BigNumber.sum(...squaredDifferences).div(numDataPoints);
        const stdDev = variance.sqrt();
        resolve({
          mean, stdDev, variance, numDataPoints,
        });
      })
      .on('error', () => reject());
  });
}
```